sdks/java/io/solr/src/test/java/org/apache/beam/sdk/io/solr/SolrIOTest.java

```diff
@@ -0,0 +1,263 @@
+    }
+  }
+
+  @Test
+  public void testSizes() throws Exception {
+    SolrIOTestUtils.insertTestDocuments(NUM_DOCS, solrClient);
+
+    PipelineOptions options = PipelineOptionsFactory.create();
+    SolrIO.Read read = SolrIO.read().withConnectionConfiguration(connectionConfiguration);
+    SolrIO.BoundedSolrSource initialSource = new SolrIO.BoundedSolrSource(read, null);
+    // can't use equal assert as Solr collections never have same size
+    // (due to internal Lucene implementation)
+    long estimatedSize = initialSource.getEstimatedSizeBytes(options);
+    LOG.info("Estimated size: {}", estimatedSize);
+    assertThat(
+        "Wrong estimated size bellow minimum",
+        estimatedSize,
+        greaterThan(SolrIOTestUtils.MIN_DOC_SIZE * NUM_DOCS));
+    assertThat(
+        "Wrong estimated size beyond maximum",
+        estimatedSize,
+        greaterThan(SolrIOTestUtils.MAX_DOC_SIZE * NUM_DOCS));
+  }
+
+  @Test
+  public void testRead() throws Exception {
+    SolrIOTestUtils.insertTestDocuments(NUM_DOCS, solrClient);
+
+    PCollection<SolrDocument> output =
+        pipeline.apply(
+            SolrIO.read()
+                .withConnectionConfiguration(connectionConfiguration)
+                .withBatchSize(101L));
+    PAssert.thatSingleton(output.apply("Count", Count.<SolrDocument>globally()))
+        .isEqualTo(NUM_DOCS);
+    pipeline.run();
+  }
+
+  @Test
+  public void testReadWithQuery() throws Exception {
+    SolrIOTestUtils.insertTestDocuments(NUM_DOCS, solrClient);
+
+    PCollection<SolrDocument> output =
+        pipeline.apply(
+            SolrIO.read()
+                .withConnectionConfiguration(connectionConfiguration)
+                .withQuery("scientist:Franklin"));
+    PAssert.thatSingleton(output.apply("Count", Count.<SolrDocument>globally()))
+            .isEqualTo(NUM_DOCS / NUM_SCIENTISTS);
+    pipeline.run();
+  }
+
+  @Test
+  public void testWrite() throws Exception {
+    List<SolrInputDocument> data = SolrIOTestUtils.createDocuments(NUM_DOCS);
+    pipeline
+        .apply(Create.of(data))
+        .apply(SolrIO.write().withConnectionConfiguration(connectionConfiguration));
+    pipeline.run();
+
+    long currentNumDocs = SolrIOTestUtils.commitAndGetCurrentNumDocs(solrClient);
+    assertEquals(NUM_DOCS, currentNumDocs);
+
+    QueryResponse response = solrClient.query(new SolrQuery("scientist:Lovelace"));
+    assertEquals(NUM_DOCS / NUM_SCIENTISTS, response.getResults().getNumFound());
+  }
+
+  @Test
+  public void testWriteWithMaxBatchSize() throws Exception {
+    SolrIO.Write write =
+        SolrIO.write()
+            .withConnectionConfiguration(connectionConfiguration)
+            .withMaxBatchSize(BATCH_SIZE);
+    // write bundles size is the runner decision, we cannot force a bundle size,
+    // so we test the Writer as a DoFn outside of a runner.
+    DoFnTester<SolrInputDocument, Void> fnTester = DoFnTester.of(new SolrIO.Write.WriteFn(write));
```

Comment:
I think DoFnTester can be used with a try-with-resources statement so you don't have to .close() it explicitly at the end.